### PR TITLE
GitHub: Cover all repo pages, un-break URL, license, and search

### LINKED ISSUE #2390: GitHub: Fix incorrect URL in embedded metadata on some repo pages
https://forums.zotero.org/discussion/89795/bug-report-zotero-connector-chrome-extension-saves-wrong-url-for-many-github-pages

### PR DESCRIPTION
**Pull request by AbeJellinek**
Fixes #2390

### CODE REVIEW

File: GitHub.js [hunk 1]
```diff
@@ -79,13 +80,24 @@ function doWeb(doc, url) {
...
 	item.url = ZU.xpathText(doc, '//meta[@property="og:url"]/@content');
+	if (url.includes('/blob/') && !item.url.includes('/blob/')) {
+		// github is doing something weird with the og:url meta tag right now -
+		// it always points to the repo root (e.g. zotero/translators), even
+		// when we're in a specific directory/file (e.g. zotero/translators/
+		// blob/master/GitHub.js). this fix (hopefully) won't stick around
+		// long-term, but for now, let's just grab the user-facing permalink
+		let permalink = attr(doc, '.js-permalink-shortcut', 'href');
+		if (permalink) {
+			item.url = 'https://' + doc.location.host + permalink;
```
Comment (dstillman) [inline GitHub.js:98]: `doc.location.host` → `github.com` — no reason not to hard-code, is there?

File: GitHub.js [hunk 2]
```diff
@@ -252,6 +262,28 @@ var testCases = [
...
+	{
+		"type": "web",
+		"url": "https://github.com/zotero/translators/blob/master/GitHub.js",
+		"items": [
+			{
+				"itemType": "computerProgram",
+				"title": "zotero/translators",
+				"creators": [],
+				"date": "2021-05-26T17:17:38Z",
+				"abstractNote": "Zotero Translators. Contribute to zotero/translators development by creating an account on GitHub.",
```
Comment (dstillman) [inline GitHub.js:275]: While we're in here, can we strip the "Contribute to…" part (which may or may not be localized)? Really should only include the description entered by the repo owner.

File: GitHub.js (hunk 1 above)
Comment (AbeJellinek) [reply GitHub.js:98]: Hmm, my reasoning was that it's one less thing to change if the translator is extended to support Enterprise installs / [an alternate domain](https://en.wikipedia.org/wiki/Censorship_of_GitHub) / some other weird case. But not *really*.

File: GitHub.js (hunk 2 above)
Comment (AbeJellinek) [reply GitHub.js:275]: Sure, I'll give it a try.

File: GitHub.js (hunk 1 above)
Comment (dstillman) [reply GitHub.js:98]: I guess that's fine.

Maybe worth including the `clipboard-copy` element, which has the absolute URL in its `value` attribute, as a backup to `.js-permalink-shortcut`?

File: GitHub.js [hunk 3]
```diff
@@ -94,7 +94,13 @@ function scrape(doc, url) {
 		// long-term, but for now, let's just grab the user-facing permalink
 		let permalink = attr(doc, '.js-permalink-shortcut', 'href');
 		if (permalink) {
-			item.url = 'https://' + doc.location.host + permalink;
+			item.url = 'https://github.com' + permalink;
+		}
+		else {
+			let clipboardCopyPermalink = attr(doc, '#blob-more-options-details clipboard-copy', 'value');
```
Comment (dstillman) [inline GitHub.js:100]: Seems like it could just be `'clipboard-copy'`

Comment (AbeJellinek) [reply GitHub.js:100]: There are a lot of `clipboard-copy`s on some pages - this issue discussion page has seven. The more specific selector makes sure that it only catches the URL one.

Comment (dstillman) [reply GitHub.js:100]: Ah, OK

**Pull request merged**